osu.Game/Overlays/BeatmapListing/BeatmapListingFilterControl.cs

```diff
@@ -186,6 +188,9 @@ private void performRequest()
                 if (sets.Count == 0)
                     noMoreResults = true;
 
+                if (CurrentPage == 0)
+                    searchControl.BeatmapSet = sets.FirstOrDefault();
```

Comment:
This **really** should have been in its own PR. Please don't mix fixes together again.